internal/shortener/shortener.go

```
@@ -2,47 +2,87 @@ package shortener
 
 import (
 	"math/rand"
-	"net/url"
 	"strings"
 )
 
-type SimpleShortener struct{}
+type Shortener interface {
+	GetShortPath(srcURL string) (string, error)
+	GetSourceURL(shortPath string) (string, error)
+	DeleteSourceURL(srcURL string) error
+	// GetCustomShortURL(source, custom string) error
+}
 
-func (s SimpleShortener) MakeShortPath() string {
-	str := generateString(5)
+type URLStorage interface {
+	Save(shortPath, srcURL string) error
+	GetShortPath(srcURL string) (string, error)
+	GetSourceURL(shortPath string) (string, error)
+	DeleteSourceURL(srcURL string) error
+}
 
-	return "/" + str
+type ShortenerService struct {
+	storage URLStorage
 }
 
-func (s SimpleShortener) ValidateURL(str string) (bool, error) {
-	u, err := url.Parse(str)
+func NewShortenerService(storage URLStorage) (*ShortenerService, error) {
```

If you don't do any blocking operation that can finish with an error then you don't need to return the error here.